### PR TITLE
feat: EVM tx details

### PR DESCRIPTION
**Pull request by evilpeach**
Fixes CFE-606, CFE-607, CFE-629, CFE-631, CFE-625, CFE-626

### CODE REVIEW

File: src/lib/app-provider/env.ts [hunk 1]
```diff
@@ -100,6 +100,7 @@ export enum CELATONE_QUERY_KEYS {
   // TX
   TX_DATA = "CELATONE_QUERY_TX_DATA",
   TX_DATA_JSON_RPC = "CELATONE_QUERY_TX_DATA_JSON_RPC",
+  COSMOS_TX_HASH_BY_EVM_TX_HASH = "CELATONE_QUERY_COSMOS_TX_HASH_BY_EVM_TX_HASH",
```
Comment (songwongtp) [inline src/lib/app-provider/env.ts:103]: put in the `// EVM` section

File: src/lib/pages/evm-tx-details/components/EvmTxDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      <Heading as="h6" variant="h6">
+        Input Data
+      </Heading>
+      {/* TODO: Can't find border color in theme */}
+      <Box
+        borderRadius="8px"
+        minHeight="100px"
+        py={4}
+        px={3}
+        border="1px solid #343445"
```
[the remark this answers is not in the document]
Comment (evilpeach) [reply src/lib/pages/evm-tx-details/components/EvmTxDetail.tsx]: I already used `TextReadonly` component.

File: src/lib/pages/evm-tx-details/components/EvmTxDetail.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { Alert, AlertDescription, Flex, Heading } from "@chakra-ui/react";
+
+import { CustomIcon } from "lib/components/icon";
+import { TextReadOnly } from "lib/components/json/TextReadOnly";
+import type { TxData, TxDataJsonRpc } from "lib/services/types";
+import type { Option } from "lib/types";
+
+import { EvmTxDetailBody } from "./EvmTxDetailBody";
+
+interface EvmTxDetailProps {
```
Comment (songwongtp) [inline src/lib/pages/evm-tx-details/components/EvmTxDetail.tsx:10]: `EvmTxMsgDetails` for consistency with cosmos tx (also filename)

File: src/lib/pages/evm-tx-details/components/EvmTxDetail.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  evmTxData: TxDataJsonRpc;
+  cosmosTxData: TxData;
+  evmDenom: Option<string>;
+}
+
+export const EvmTxDetail = ({
```
Comment (songwongtp) [inline src/lib/pages/evm-tx-details/components/EvmTxDetail.tsx:16]: ditto

File: src/lib/pages/evm-tx-details/components/EvmTxDetailBody.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import type { Option } from "lib/types";
+import { getEvmMethod } from "lib/utils";
+
+import {
+  EvmTxCreateContract,
+  EvmTxTransfer,
+  EvmTxTransferErc20,
+} from "./evm-tx-method";
+
+interface EvmTxDetailProps {
```
Comment (songwongtp) [inline src/lib/pages/evm-tx-details/components/EvmTxDetailBody.tsx:15]: ditto

File: src/lib/pages/evm-tx-details/components/EvmTxDetailBody.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+  evmTxData: TxDataJsonRpc;
+  evmDenom: Option<string>;
+}
+
+export const EvmTxDetailBody = ({ evmTxData, evmDenom }: EvmTxDetailProps) => {
```
Comment (songwongtp) [inline src/lib/pages/evm-tx-details/components/EvmTxDetailBody.tsx:20]: ditto

File: src/lib/pages/evm-tx-details/components/EvmTxHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+              )}
+            </Flex>
+          )}
+        </Flex>
+        {!isMobile && (
+          <Button
+            variant="ghost-gray"
+            rightIcon={<CustomIcon name="launch" boxSize={3} m={0} />}
+            onClick={() => {
+              trackUseViewJSON("tx_page_transaction_hash");
```
Comment (songwongtp) [inline src/lib/pages/evm-tx-details/components/EvmTxHeader.tsx:72]: `evm_tx_...`

File: src/lib/pages/evm-tx-details/components/EvmTxHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+                  color="success.main"
+                />
+                <Text variant="body2" color="success.main">
+                  Success
+                </Text>
+              </>
+            )}
+          </Flex>
+        )}
+        {cosmosTxData.timestamp ? (
```
Comment (songwongtp) [inline src/lib/pages/evm-tx-details/components/EvmTxHeader.tsx:145]: I don't think `timestamp` is optional

File: src/lib/services/tx/index.ts [hunk 9]
```diff
@@ -157,7 +161,7 @@ export const useTxsByAddress = (
       isMove,
     ],
     async () => {
-      if (!address) throw new Error("No user address");
+      if (!address) throw new Error("addess is undefined (useTxsByAddress)");
```
Comment (songwongtp) [inline src/lib/services/tx/index.ts:164]: address

File: src/lib/services/tx/index.ts [hunk 10]
```diff
@@ -563,4 +567,33 @@ export const useTxDataJsonRpc = (evmTxHash: string) => {
...
+    },
+    {
+      retry: false,
+      refetchOnWindowFocus: false,
+      enabled: evm.enabled && !!evm.jsonRpc,
+    }
+  );
+};
+
+export const useCosmosTxDataByEvmTxHash = (evmTxHash: string) => {
```
Comment (songwongtp) [inline src/lib/services/tx/index.ts:593]: maybe move this to `src/lib/pages/evm-tx-details/data.ts`

**Review by songwongtp — APPROVED** [review]

**Pull request merged**